cmd/kube-proxy/app/server_others.go

```diff
@@ -317,6 +347,76 @@ func newProxyServer(
 	}, nil
 }
 
+func getDetectLocalMode(config *proxyconfigapi.KubeProxyConfiguration) proxyconfigapi.LocalMode {
+	switch config.DetectLocalMode {
+	case proxyconfigapi.LocalModeClusterCIDR:
+		return config.DetectLocalMode
+	default:
+		klog.V(4).Info("Defaulting detect local mode to ", string(proxyconfigapi.LocalModeClusterCIDR))
+		return proxyconfigapi.LocalModeClusterCIDR
+	}
+}
+
+func getIPTablesLocalDetector(mode proxyconfigapi.LocalMode, config *proxyconfigapi.KubeProxyConfiguration, ipt utiliptables.Interface) (localDetector utiliptables.LocalTrafficDetector, err error) {
```

Comment:
This seems identical to the ipvs form - why 2 functions, rather than one function in `pkg/proxy/util/iptables` ?

And for that matter, why not push these functions down into the proxy modules, if they are highly coupled enough to need different implementations?

Reply:
Because of the different types. But given above information of refactoring it to one, can simplify it further.